plugins/packages/polygon/main.go

```diff
@@ -355,30 +371,55 @@ func Migrate(db *sql.DB, chainid uint64) error {
 			if _, err := dbtx.Exec(statement); err != nil {
 				dbtx.Rollback()
 				log.Warn("Failed to insert statement polygong migration v2", "err", err.Error())
-				continue
+				return nil
 			}
-
-			
-			if number%500 == 0 {
+			if number <= terminus {
+				if number%500 == 0 { 
+					if err := dbtx.Commit(); err != nil {
+						log.Error("Failed to commit statements in loop, polygon plugin", "blockNumber", number, "err", err.Error())
+						return nil
+					}
+					log.Info("blocks migration in progress", "blockNumber", number)
+					dbtx, err = db.BeginTx(context.Background(), nil)
+					if err != nil {
+						log.Error("Error creating a transaction in loop, polygon plugin", "err", err.Error())
+						return nil
+					}
+				} 
+			}
+			if number == highestBlock {
 				if err := dbtx.Commit(); err != nil {
-					log.Warn("Failed to insert statement polygon plugin", "blockNumber", number, "err", err.Error())
-					continue
-				}
-				log.Info("blocks migration in progress", "blockNumber", number)
-				dbtx, err = db.BeginTx(context.Background(), nil)
-				if err != nil {
-					log.Warn("Error creating a transaction polygon plugin", "err", err.Error())
+					log.Error("Failed to insert statements at terminus, polygon plugin", "blockNumber", number, "err", err.Error())
+					return nil
 				}
+				log.Info("polygon migration v2 finished on block", "blockNumber", number)
+			}
+		}
+			if _, err := db.Exec("UPDATE bor.migrations SET version = 2;"); err != nil {
+				log.Error("polygon migrations v2 error", "err", err.Error())
+				return nil
 			}
+	}
+	
 
+	if schemaVersion < 3 {
```

Comment:
These don't belong in bor migrations - we should update blocks and transactions migrators to add their respective idnexes.